### PR TITLE
Allow GTK style enum naming

### PR DESCRIPTION
**Pull request by robert-ancell**
Remove the linting overrides for enum names and set this in the clang-format file.

### CODE REVIEW

**Comment by robert-ancell** [comment]
I didn't remove the lint disabling in the public headers, as I'm not sure if this would affect the apps that use them to generate warnings - does anyone know if this is appropriate?

File: shell/platform/linux/.clang-tidy [hunk 1]
```diff
@@ -1,5 +1,7 @@
 InheritParentConfig: true
 
 # EnumCastOutOfRange warns about some common usages of GTK macros
+# EnumCase allows GTK style enum values
 Checks: >-
   -clang-analyzer-optin.core.EnumCastOutOfRange
+  -readability-identifier-naming.EnumCase
```
Comment (cbracken) [inline shell/platform/linux/.clang-tidy:7]: Would this work?
```
CheckOptions:
  - key: readability-identifier-naming.EnumConstantCase
    value: "UPPER_CASE"
```

**Comment by cbracken** [comment]
> I didn't remove the lint disabling in the public headers, as I'm not sure if this would affect the apps that use them to generate warnings - does anyone know if this is appropriate?

I'd _assume_ the onus would be on users to opt their tooling out of linting third-party headers. That doesn't mean people aren't doing this, but I'd imagine that they'd run into this with other libraries as well.

**Comment by stuartmorgan** [comment]
test-exempt: enforced by configuration change

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Pull request merged**